generators/app/templates/src/main/java/package/service/_ElasticsearchIndexService.java

```diff
@@ -141,11 +147,30 @@ public void reindexAll() {
         }
         elasticsearchTemplate.putMapping(entityClass);
         if (jpaRepository.count() > 0) {
-            try {
-                Method m = jpaRepository.getClass().getMethod("findAllWithEagerRelationships");
-                elasticsearchRepository.save((List<T>) m.invoke(jpaRepository));
-            } catch (Exception e) {
-                elasticsearchRepository.save(jpaRepository.findAll());
+            // set up a list of relationships that should be eagerly loaded.
+            // if a JHipster entity field is a Set, it is a relationship
```

Comment:
Update this comment to `// if a JHipster entity field is the owner side of a many-to-many relationship, it should be loaded manually`